### ISSUE TITLE
Minimum/maximum supported URL length for API queries is not documented (or guaranteed fwiw), provide it via capabilities call

### ISSUE THREAD

**Issue by simonpoole** [openstreetmap/openstreetmap-website#4850]
### Problem

A number of the API queries can have a variable number of parameters, for example both the element and user multi-fetch API. However no minimum and/or maximum supported length of the URL is documented, outside of what somebody found out by experimentation see https://wiki.openstreetmap.org/wiki/API_v0.6#Multi_fetch:_GET_/api/0.6/[nodes|ways|relations]?#parameters

Normally I would have suggested to simply document what both the maintainers and OSM ops consider they should support, but given reality, I believe it would be better to provide the current value via the _capabilities_  call so that API users can dynamically adapt to whatever is currently supported.

**Comment by tomhughes** [comment]
The trouble is I'm not sure there's any good way to know what the limit is, and it's certainly not something we can guarantee as the web site developers as it may well in part come down to what the web server and/or application server used to run the site have as limits.

**Comment by simonpoole** [comment]
> The trouble is I'm not sure there's any good way to know what the limit is, and it's certainly not something we can guarantee as the web site developers as it may well in part come down to what the web server and/or application server used to run the site have as limits.

That is why I'm suggesting providing a value via the capabilities API, because then you can just adjust the reported value when the inevitable breakage happens instead of editor devs having to rush out new versions of their apps.

**Comment by AntonKhorev** [comment]
You can't report Internet Explorer's limits via the capabilities API.

I've been doing requests with ~700 elements.

**Comment by bhousel** [comment]
> You can't report Internet Explorer's limits via the capabilities API.

Yes, sorry if it wasn't clear from my comment - I don't think URL limits belong in the capabilities API.

If there is a "maximum number of entity IDs allowed in multi fetch GET", this would be a reasonable limit to report in capabilities, but URL length is a separate thing.

**Comment by mmd-osm** [comment]
FWIW: CGImap has no explicit limit, it will happily process whatever the webserver allows in terms of max request size.

One thing I'm missing in this discussion in the total size of the http header fields, which could also result in "431 Request Header Fields Too Large". If for whatever reason someone is sending lots of large cookies along with their API request, this might have some impact on  the max number of OSM objects they can request.

**Comment by simonpoole** [comment]
The problem is that the max possible number of entities per call depends -both- on the API implementation and the maximum URL length supported by whatever is in front of it.

And yes last week somebody managed to exceed the limit and get back a 414 error from Apache because the max URL length was exceeded.
